### PR TITLE
Adds a tool that filters leading slashes from topic names in bags.

### PR DESCRIPTION
**Pull request by SirVer**
PAIR=@cschuet

### CODE REVIEW

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""A simple tool to remove leading slashes from bags."""
+
+import rosbag
+import argparse
+
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: Python style: two empty lines between top-level definitions.

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      msg.header.frame_id = msg.header.frame_id[1:]
+  if hasattr(msg, "child_frame_id"):
+    if msg.child_frame_id.startswith("/"):
+      msg.child_frame_id = msg.child_frame_id[1:]
+  if hasattr(msg, "transforms"):
+    for transform_msg in msg.transforms:
+      rewrite(transform_msg)
+
+def main():
+  o = parse_args()
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: Style: Eschew abbreviations?

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: `python3` should be possible, right?

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+"""A simple tool to remove leading slashes from bags."""
+
+import rosbag
+import argparse
+
+def parse_args():
+  p = argparse.ArgumentParser(description=
+    "Removes leading slashes from topic names in a bag."
+  )
+  p.add_argument("input", type=str, help = "Input bag")
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: No whitespace around `=` in `help="Input bag"`

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""A simple tool to remove leading slashes from bags."""
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: ...`leading slashes from frame names`?

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+import rosbag
+import argparse
+
+def parse_args():
+  p = argparse.ArgumentParser(description=
+    "Removes leading slashes from topic names in a bag."
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: typo: `topic` -> `frame`

File: cartographer_ros/scripts/remove_leading_slashes.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  if hasattr(msg, "child_frame_id"):
+    if msg.child_frame_id.startswith("/"):
+      msg.child_frame_id = msg.child_frame_id[1:]
+  if hasattr(msg, "transforms"):
+    for transform_msg in msg.transforms:
+      rewrite(transform_msg)
+
+def main():
+  o = parse_args()
+  with rosbag.Bag(o.input[:-4] + '.filtered.bag', 'w') as outbag:
```
Comment (wohe) [inline cartographer_ros/scripts/remove_leading_slashes.py]: `os.path.splitext(o.input)[0]` or so?

File: cartographer_ros/scripts/remove_leading_slashes.py (hunk 3 above)
Comment (SirVer) [reply cartographer_ros/scripts/remove_leading_slashes.py]: dunno. the ROS packages seem to be all python2 on my system.

**Pull request merged**